mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -696,6 +800,63 @@ def log_param(self, run_id, param):
                 else:
                     raise
 
+    def _log_params(self, run_id, params):
+        if not params:
+            return
+        # eliminate duplicate Param entities
+        param_instances = {}
+        for param in params:
+            if param.key not in param_instances:
+                param_instances[param.key] = SqlParam(
+                    run_uuid=run_id, key=param.key, value=param.value
+                )
+        param_instances = list(param_instances.values())
```

Comment:
I think we can replace this with 

```
params = set(params)
param_instances = [SqlParam(run_uuid=run_id, key=param.key, value=param.value)]
```

Reply:
I am not sure why I choose this way of doing this, but I think I wanted to make sure that we keep the order of params while making them distinct, since set does not preserve the order

Reply:
Oh, the order matters in tags, so it made me think so :) applied it over here